cmd/expired-authz-purger/main.go

```diff
@@ -95,34 +96,50 @@ func (p *expiredAuthzPurger) purge(table string, yes bool, purgeBefore time.Time
 		}
 	}
 
+	sem := make(chan bool, parallelism)
```

Comment:
This seems like a slightly complicated way of attaining parallelism, why not go for the more standard `sync.WaitGroup` with a work `chan` approach?

i.e. while slightly longer I think this is easier to understand and follows the general style we use elsewhere
```
wg := new(sync.WaitGroup()
work := make(chan string, len(ids))
for _, id := range ids {
    work <- id
}
close(work)
for i := 0; i < parallelism; i++ {
    wg.Add(1)
    go func() {
        defer wg.Done()
        for id := range work {
            ...
        }
    }()
}
wg.Wait()
```

Reply:
Sounds good, will do. I was originally going to write this style, but I got frustrated writing this same loop all the time - it seems like a language designed around concurrency should have a cleaner primitive. In particular, why use two synchronizations constructs (WaitGroup and channels) for one simple task? I found [this page](http://jmoiron.net/blog/limiting-concurrency-in-go/), which suggested the concurrency pattern implemented here, copied from the `net` package.

All that said, local style rules, so I'm going to update this to match local style. :-)